### PR TITLE
core: Removed remaining occurences of memmove/memcpy of Particle, and…

### PR DESCRIPTION
**Pull request by fweik**
… added a dummy overload for memmove and memcpy to make such calls cause a compile time error.

Fixes # segfaults
Closes #1333. Closes #1285.

- Hopefully fixes all the problems with regard to the Particles
- Enables parallel tests on travis
- Python tests work in Debug build
- Most of the python tests work in parallel

~~EK is sill broken though... (See #1337)~~

### CODE REVIEW

**Review by hmenke — APPROVED** [review]

**Comment by fweik** [comment]
Performance for a 1000 part WCA system went down from
0.233 +/- 0.002 ms to 0.244 +/- 0.002 ms per time step.

**Comment by KaiSzuttor** [comment]
Why are the tests failing on mac?

**Comment by KaiSzuttor** [comment]
coulomb_cloud_wall.py hangs if executed with > 1 cores.

**Comment by fweik** [comment]
The coulomb test hangs due to scafacos, ~~I don't think this is caused by espresso.~~ Without it this works find (as for example on travis). The scafacos issue will be fixed elsewhere.

**Comment by fweik** [comment]
~~~The mac issue seems to be caused by a too old cmake version for the boost version. (https://stackoverflow.com/questions/42123509/cmake-finds-boost-but-the-imported-targets-not-available-for-boost-version). I'm not going to deal with this.~~~

File: src/core/communication.cpp [hunk 1]
```diff
@@ -257,32 +259,27 @@ void mpi_init(int *argc, char ***argv) {
   }
 #endif
 
-  MPI_Init(argc, argv);
+  mpi_env =
+      Utils::make_unique<boost::mpi::environment>(*argc, *argv);
```
Comment (KaiSzuttor) [inline src/core/communication.cpp]: On mac os, i get a segfault on exactly this line:
```
thread #2: tid = 0x1dfaf, 0x00000001023baeeb libmpi.12.dylib`MPI_Comm_set_errhandler + 219, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x8)
  * frame #0: 0x00000001023baeeb libmpi.12.dylib`MPI_Comm_set_errhandler + 219
    frame #1: 0x00000001024ca494 libboost_mpi-mt.dylib`boost::mpi::environment::environment(int&, char**&, bool) + 84
    frame #2: 0x00000001026d4deb libEspressoCore.4.dylib`mpi_init(argc=0x0000000000000000, argv=0x0000000000000000) + 651 at communication.cpp:268
    frame #3: 0x00000001019cabec _init.so`::init_init() + 1948 at _init.cpp:1090
    frame #4: 0x00000001001662ec Python`_PyImport_LoadDynamicModule + 140
    frame #5: 0x0000000100164cef Python`import_submodule + 271
    frame #6: 0x0000000100164778 Python`load_next + 280
    frame #7: 0x000000010016380e Python`PyImport_ImportModuleLevel + 1230
    frame #8: 0x000000010013c7c7 Python`builtin___import__ + 135
    frame #9: 0x000000010009e203 Python`PyObject_Call + 99
    frame #10: 0x0000000100147429 Python`PyEval_EvalFrameEx + 22425
    frame #11: 0x0000000100141947 Python`PyEval_EvalCodeEx + 2119
    frame #12: 0x00000001001410f6 Python`PyEval_EvalCode + 54
    frame #13: 0x00000001001621a1 Python`PyImport_ExecCodeModuleEx + 241
    frame #14: 0x0000000100165273 Python`load_source_module + 1059
    frame #15: 0x00000001001655ef Python`load_package + 303
    frame #16: 0x0000000100164cef Python`import_submodule + 271
    frame #17: 0x0000000100164778 Python`load_next + 280
    frame #18: 0x00000001001637b2 Python`PyImport_ImportModuleLevel + 1138
    frame #19: 0x000000010013c7c7 Python`builtin___import__ + 135
    frame #20: 0x000000010009e203 Python`PyObject_Call + 99
    frame #21: 0x0000000100147429 Python`PyEval_EvalFrameEx + 22425
    frame #22: 0x0000000100141947 Python`PyEval_EvalCodeEx + 2119
    frame #23: 0x00000001001410f6 Python`PyEval_EvalCode + 54
    frame #24: 0x0000000100170824 Python`PyRun_FileExFlags + 164
    frame #25: 0x000000010017035e Python`PyRun_SimpleFileExFlags + 702
    frame #26: 0x0000000100186b1d Python`Py_Main + 2925
    frame #27: 0x00007fffbbe11235 libdyld.dylib`start + 1
    frame #28: 0x00007fffbbe11235 libdyld.dylib`start + 1
```

**Comment by fweik** [comment]
@KaiSzuttor should be fixed. Was that a debug build or a normal one?

**Comment by KaiSzuttor** [comment]
It was a debug build. Your change did not fix it.

**Comment by fweik** [comment]
travis is happy now, @KaiSzuttor can you please recheck that you still see the issue? Otherwise, ready for review. @RudolfWeeber

**Comment by KaiSzuttor** [comment]
I don't understand why it still crashes, if I instead call MPI_init(NULL, NULL) it works. Maybe I have issues with libs on my system and we can ignore this issue.

**Comment by fweik** [comment]
That is especially odd because this is exactly what boost.mpi calls
internally.

**Comment by KaiSzuttor** [comment]
I just built boost from source and it works. Maybe another compiler-dependency.

**Comment by RudolfWeeber** [comment]
Can be merged.

**Pull request merged**